pandas/core/reshape.py

```diff
@@ -1121,6 +1138,18 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     1  0  1    0
     2  0  0    1
 
+    >>> pd.get_dummies(s1, fill_value=np.nan)
+         a    b
+    0    1    0
+    1    0    1
+    2  NaN  NaN
+
+    >>> pd.get_dummies(s1, fill_value=np.nan, dummy_na=True)
+         a    b  NaN
+    0    1    0    0
+    1    0    1    0
+    2  NaN  NaN    1
```

Comment:
> Yes, you can do that, but I think fill_value=NaN is a poor default for the case of dummy_na=True.

Agreed. I'm struggling to think of a reason you would ever want `fill_value=np.nan, dummy_na=True`... But I don't think we should raise an exception in this case, if the user specifies those arguments.